### ISSUE TITLE
Binary segmentation on a custom dataset

### ISSUE THREAD

**Issue by cspearl** [holyseven/PSPNet-TF-Reproduce#31]
Hi, 
I am new to segmentation and trying to use this repo to train the network from scratch on a custom dataset having only 2 classes (binary segmentation) and so if you could please guide me as to where do i need to make the changes in the code.
Also is it possible to train if a dataset has images in 4 channels or 1 channel and not 3.
A help would be highly appreciated.
Thanks!!

**Comment by holyseven** [comment]
Hi,

> 2 classes

Change the `num_classes` when calling [PSPNet](https://github.com/holyseven/PSPNet-TF-Reproduce/blob/master/model/pspnet_mg.py#L9).

> Input images' channels.

I think you don't need to change anything in the code. But you can't use the pre-trained ImageNet weights, defined in [FLAGS.fine_tune_filename](https://github.com/holyseven/PSPNet-TF-Reproduce/blob/master/run.py#L125).

**Comment by cspearl** [comment]
While studying the code I found out for the databases tf.decode_jpg or png is used but my dataset has images of type .tif (1024*1024) . I saw that there is a tfio.experimental.decode_tif method but that is for a recent version of tf . I dont know whether tfio will work with this code.
How should I solve this problem?

**Comment by holyseven** [comment]
I'm not so familiar with tfio either. Sorry I can't help you about this problem.

Try some demo codes to verify whether the code can read tif images. Or just convert .tif images to .png .jpg images with other tools.

**Comment by cspearl** [comment]
Do you know exactly what tf.decode does? Then probably I could use some other function for tiff images?

**Comment by holyseven** [comment]
`tf.decode` decodes the images from files to arrays/tensors. If I remember correctly, this repo uses tensorflow1.4, which has separate apis for different formats of images. Not sure if there exists `tf.decode_tiff` in 1.4. But if you use higher version of tensorflow, I think all apis are aggregated to a single one, which decodes all formats of images. That could decode tiff images probably.

**Comment by cspearl** [comment]
Yes i checked version 1 of tf does not support tif neither does v2. But there is an option in experimental tf-io as I had said but I will try some other method thank you.
Can you help with images in 4 channels- RGB(3)+NIR(1) so there are two separate images. How should I pass them onto the reader like to fuse the two seperate matrices of the images H*W*3 and H*W*1?

**Comment by holyseven** [comment]
You can prepare a function for reading images, NIRs, and labels, similar to https://github.com/holyseven/PSPNet-TF-Reproduce/blob/master/database/reader.py#L28
